src/test/kotlin/org/rust/lang/core/type/RsGenericExpressionTypeInferenceTest.kt

```diff
@@ -1431,4 +1431,34 @@ class RsGenericExpressionTypeInferenceTest : RsTypificationTestBase() {
             a;
         } //^ X
     """)
+
+    fun `test field of parent with deref`() = testExpr("""
```

Comment:
Let's rename it to "test field type substitution after deref".
Also, this test can be written much more compact. We can write structs in a single line `struct S1<A> { field1: A }`, remove field2 from S2, remove `deref` method from Deref trait and impl.